### PR TITLE
docs: improve search 1/

### PR DESCRIPTION
**Pull request by yuhan**
### Summary & Motivation
What went wrong
- Changelog and migration shouldn't be included when indexing - this PR explicitly stops indexing these in [1].
- Changelog and migration and cloud site shouldn't be included in sitemap (docs.dagster.io/sitemap.xml) which is the starting point of the indexing script too - this PR removes these entries from the xml generation script in [2].
- Apidocs weren't indexed recently because the html is now using "section" id instead of "section" class - this PR changes how the search indexing script finds api docs in [3].
- Section title wasn't properly marked as class "DocSearch-lvl1" (which is what the search indexing script identifies as the section name to fill) so missing that class resulted in search results falling back to the default value "section" (see second bullet on issue DREL-296 and screenshot below) - this PR adds that class back in [4]
  image
  * **once we land this, reindexing should fill out all the "section" in the screenshot with correct title names**

### How I Tested These Changes
- Reran the search indexing script locally which 1) stopped crawling changelog and migration (changelog and migration entries no longer show up in the docs search) and 2) started to crawl api docs (apidocs show up in the search).
- Wait to land this to prod so we can rerun the script again to pick up the section title fix.

### CODE REVIEW

File: docs/config.json [hunk 1]
```diff
@@ -11,7 +11,9 @@
   "stop_urls": [
     "https://docs.dagster.io/_modules/",
     "https://docs.dagster.io/master/",
-    "https://docs.dagster.io/\\d+\\.\\d+.\\d+/"
+    "https://docs.dagster.io/\\d+\\.\\d+.\\d+/",
+    "https://docs.dagster.io/changelog",
+    "https://docs.dagster.io/migration"
```
Comment (yuhan) [inline docs/config.json:16]: [1]

File: docs/config.json [hunk 2]
```diff
@@ -41,12 +43,12 @@
...
-      "lvl3": ".section h3",
-      "lvl4": ".section h4",
-      "lvl5": ".section h5, .section .rubric, .section dt",
-      "text": ".section p, .section li"
+      "lvl1": "section h1",
+      "lvl2": "section h2",
+      "lvl3": "section h3",
+      "lvl4": "section h4",
+      "lvl5": "section h5, section .rubric, section dt",
+      "text": "section p, section li"
```
Comment (yuhan) [inline docs/config.json:51]: [3]

File: docs/next/components/Sidebar.tsx [hunk 3]
```diff
@@ -66,7 +66,9 @@ const MenuItem = React.forwardRef<
         )}
         <span
           className={cx({
+            // Map content hierarchy to levels for docs search
             "DocSearch-lvl0": lvl === 0 && match,
+            "DocSearch-lvl1": lvl === 1 && match,
```
Comment (yuhan) [inline docs/next/components/Sidebar.tsx:71]: [4]

File: docs/next/pages/sitemap.xml.tsx [hunk 4]
```diff
@@ -11,10 +11,9 @@ const createSitemap = (host, routes) =>
 const Sitemap = () => {};
 
 Sitemap.getInitialProps = ({ res, req }) => {
-  const routes = latestAllPaths().map(
+  const routes = latestAllVersionedPaths().map(
```
Comment (yuhan) [inline docs/next/pages/sitemap.xml.tsx:14]: [2]

**Review by sryza — APPROVED** [review]
Really great to get this stuff fixed. Docs search is so important.

**Pull request merged**